src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -397,6 +473,11 @@ void RewriteONNXForZHighPass::runOnOperation() {
     return true;
   });
 
+  target.addDynamicallyLegalOp<ONNXConvOp>([](ONNXConvOp op) {
+    return !op.pads() || !hasNonZeroInArrayAttr(op.pads().value()) ||
+           isSuitableForZDNN<ONNXConvOp>(op);
```

Comment:
Is it enough to use only `isSuitableForZDNN<ONNXConvOp>(op)`? 

Looking at this It seems there is the third case where we cannot run directly on NNPA or cannot do padding for NNPA. In summary, it looks to me there are three cases:
- Conv ops that can run on NNPA directly.
- Conv ops that need to pad to run on NNPA.
- Conv ops do not belong to the above 2 cases and they still run on CPU.

I thought we only had the first two cases. Could you confirm that? Thanks!

Reply:
@tungld Thank you for your comments.

Let me explain my understanding.

This rule should be applied only when
(Condition A) Conv ops cannot run on NNPA before applying the rule <AND>
(Condition B) Conv ops can run on NNPA after applying the rule

And
(Condition A) The "(IsConvolutionOpNotSuitableForZDNN:$res)"  condition can be used.
(Condition B) After applying the rule, autopad becomes "VALID", which can be executed on NNPA always.

So that the "(IsConvolutionOpNotSuitableForZDNN:$res)"  condition is enough, as you described.

It's fine, but the current "ConvOpPaddingPattern" rule works onnx.Conv with the non-zero pads attribute only,
because it generates onnx.Pad with the specified "pads" attribute.
So that I supposed that the pattern need not be applied when the pads are not specified or zeros, and
the " !op.pads() || !hasNonZeroInArrayAttr(op.pads().value()" condition is used to stop applying the rule.

I am happy if you confirm my understanding. Thank you.

Reply:
Thank you for the useful comments.

Let me check the three cases according to pads attribute.
- "VALID" and "SAME_UPPER" cases can be supported on NNPA directly always.
- "SAME_LOWER" case is not supported on NNPA at all at this point.
- "NOTSET" case can be supported on NNPA on directly or with cpu-side padding.

The "SAME_LOWER" case may be the third case (with the current implementation).

Reply:
Updated the rule to support all onnx.Conv on NNPA, and remove all conditions except "isSuitableForZDNN<ONNXConvOp>(op)" from "addDynamicallyLegalOp<ONNXConvOp>" in RewriteONNXForZHigh.cpp